common/build/eslint-plugin-fluid/src/test/example/no-unchecked-record-access/nullableIndexedRecord.ts

```diff
@@ -0,0 +1,59 @@
+	return record.a; // defect: Returning index property 'a' as string or null variable should be caught, 'a' might be undefined
+}
+
+function recordAFnExpectsStringOrUndefinedOrNull(
+	record: NullableIndexSignatureType,
+): string | undefined | null {
+	return record.a; // ok: Returning index property 'a' to string or undefined or null variable, 'a' might be undefined
+}
+
+function AFnExpectsStringOrNull(a: string | null): string | null {
+	return a;
+}
+
+function AFnExpectsStringOrUndefinedOrNull(
+	a: string | undefined | null,
+): string | undefined | null {
+	return a;
+}
+
+AFnExpectsStringOrNull(nullableIndexedRecord.a); // defect: Passing index property 'a' to a function without having type undefined
+AFnExpectsStringOrUndefinedOrNull(nullableIndexedRecord.a); // ok: Passing index property 'a' to a function that accepts undefined is fine
+
+/*
+ * Variable Assignments
+ */
+
+const aExpectingStringOrNull: string | null = nullableIndexedRecord.a; // defect: Assigning index property 'a' to string or null variable, 'a' might not be present, type should include an undefined type
+let aLetExpectingStringOrNull: string | null = nullableIndexedRecord.a; // defect: Assigning index property 'a' to string or null variable, 'a' might not be present, either the index signature type should include an undefined type or the variable declaration should be changed to string | null | undefined
+const aExpectingStringOrNullOrUndefined: string | null | undefined = nullableIndexedRecord.a; // ok: Assigning index property 'a' to string or null or undefined variable is fine, 'a' might not be present
+let aLetExpectingStringOrNullOrUndefined: string | null | undefined = nullableIndexedRecord.a; // ok: Assigning index property 'a' to string or null or undefined variable is fine, 'a' might not be present
+
+const aImplicitType = nullableIndexedRecord.a; // defect: Index property without an explicit undefined can not be assigned to an inferred type
+
+let aLetExpectingStringOrNullAfterVariableDeclaration: string | null;
+aLetExpectingStringOrNullAfterVariableDeclaration = nullableIndexedRecord.a; // defect: Assigning index property 'a' to string or null variable should report an error, 'a' might not be present, either the index signature type should include an undefined type or the variable declaration should be changed to string | null | undefined
+let aLetExpectingStringOrNullOrUndefinedAfterVariableDeclaration: string | null | undefined;
+aLetExpectingStringOrNullOrUndefinedAfterVariableDeclaration = nullableIndexedRecord.a; // ok: Assigning index property 'a' to string or null or undefined variable, 'a' might not be present
+
+AFnExpectsStringOrNull(aImplicitType); // ok: Passing index property 'a' to a function without having type undefined should report a defect but it doesnt, instead we have disallowed implicit type assignment from index signature type without an explicit undefined
```

Comment:
Keep this with the related code. It should be directly under `const aImplicitType = nullableIndexedRecord.a;` just like in the indexedRecord case. And just like the indexed record case there should be an overall tsc comment.

Reply:
yes makes sense, updated